ambry-router/src/test/java/com.github.ambry.router/PutManagerTest.java

```diff
@@ -614,23 +617,25 @@ private CountDownLatch submitPut()
       throws Exception {
     final CountDownLatch doneLatch = new CountDownLatch(requestAndResultsList.size());
     // This check is here for certain tests (like testConcurrentPuts) that require using the same router.
-    if (router == null || !router.isOpen()) {
+    if (instantiateNewRouterForPuts) {
```

Comment:
The other change you made is the only right one I think. It will expose the scenarios where the `Router` was closed unexpectedly (either because of a bug or because of wrong expectations). This change will now mask any bugs if they exist (i.e. things that should not have closed the `Router` but did)